src/zyre.c

```diff
@@ -835,62 +884,105 @@ zyre_test (bool verbose)
         zyre_stop (node4);
 
         // give things a chance to settle...
-        zclock_sleep (500);
+        zclock_sleep (250);
+
+        zyre_destroy(&node3);
+        zyre_destroy(&node4);
+
+        zcert_destroy(&node3_cert);
+        zcert_destroy(&node4_cert);
 
         // test gossip
+        if (verbose)
+            zsys_debug ("----------------TESTING GOSSIP----------------");
 
-        zsys_debug ("----------------TESTING GOSSIP----------------");
+        zyre_t *node5 = zyre_new ("node5");
+        zyre_t *node6 = zyre_new ("node6");
 
-        const char *gossip_cert = zcert_public_txt (node3_cert);
+        assert (node5);
+        assert (node6);
 
-        zyre_gossip_bind(node3, "tcp://*:9001");
-        zyre_gossip_connect_curve(node4, gossip_cert, "tcp://127.0.0.1:9001");
+        if (verbose) {
+            zyre_set_verbose (node5);
+            zyre_set_verbose (node6);
+        }
 
-        zyre_start(node3);
-        zsock_wait(node3);
-        zyre_start(node4);
-        zsock_wait(node4);
+        // if it takes more than 10s, something probably went terribly wrong
+        zsock_set_rcvtimeo(node5->inbox, 10000);
+        zsock_set_rcvtimeo(node6->inbox, 10000);
 
-        zyre_join (node3, "GLOBAL");
-        zyre_join (node4, "GLOBAL");
+        zcert_t *node5_cert = zcert_new ();
+        zcert_t *node6_cert = zcert_new ();
 
-        zclock_sleep (500);
+        assert (node5_cert);
+        assert (node6_cert);
 
-        zyre_shouts (node3, "GLOBAL", "Hello, World");
+        zyre_set_zcert(node5, node5_cert);
+        zyre_set_zcert(node6, node6_cert);
+
+        zyre_set_header(node5, "X-PUBLICKEY", "%s", zcert_public_txt(node5_cert));
+        zyre_set_header(node6, "X-PUBLICKEY", "%s", zcert_public_txt(node6_cert));
+
+        const char *gossip_cert = zcert_public_txt (node5_cert);
+
+        // TODO- need to add zyre_gossip_port functions to get port from gossip bind(?)
```

Comment:
yes that would be good, otherwise there will be random failures in the tests

Reply:
aye- didn't know if there was an easier way to just access zyre->gossip actor for testing, didn't see one. will follow up with more PRs as we go for this.

Reply:
great, thanks!